java/opendcs/src/main/java/decodes/tsdb/IntervalIncrement.java

```diff
@@ -34,286 +15,305 @@
-	 */
-	public static IntervalIncrement[] parseMult(String s)
-		throws NoSuchObjectException
-	{
-		ArrayList<IntervalIncrement> aii = new ArrayList<IntervalIncrement>();
-		Matcher iimatcher = iipattern.matcher(s);
-		while(iimatcher.find())
-		{
-			if (iimatcher.groupCount() != 2)
-			{
-				String msg = "Invalid IntervalIncrement string '" + s + "'";
-				Logger.instance().warning(msg);
-				throw new NoSuchObjectException(msg);
-			}
-			String count = iimatcher.group(1);
-			int icount = -1;
-			try { icount = Integer.parseInt(count); }
-			catch(NumberFormatException ex)
-			{
-				// Shouldn't happen, matcher already guarantees it is digits.
-				String msg = "Bad count in Interval Increment String '"
-					+ s + "': " + ex;
-				Logger.instance().warning(msg);
-				throw new NoSuchObjectException(msg);
-			}
-			int calConst = str2CalConst(iimatcher.group(2));
-			if (calConst == -1)
-			{
-				String msg = "Bad time-interval string '"
-					+ iimatcher.group(2) + "' in string '" + s + "'";
-				Logger.instance().warning(msg);
-				throw new NoSuchObjectException(msg);
-			}
-			aii.add(new IntervalIncrement(calConst, icount));
-		}
-		if (aii.size() == 0)
-			throw new NoSuchObjectException("No valid intervals found in '" + s + "'");
-		return aii.toArray(new IntervalIncrement[aii.size()]);
-	}
+    /**
+     * Parse strings like "5 hour", "5h", "15m", "1 month".
+     * String must be a digit, optionally followed by whitespace, followed
+     * by a word representing the interval.
+     * @param s the string.
+     * @return corresponding IntervalIncrement or ONE_HOUR if parse error.
+     */
+    public static IntervalIncrement parse(String s)
+    {
+        if (s == null)
+        {
+            Logger.instance().warning("IntervalIncrement.parse called with null    -- using 1 hour");
+            return ONE_HOUR;
+        }
 
-	public String toString()
-	{
-		switch(calConstant)
-		{
-		case Calendar.YEAR: return "(" + count + " YR)";
-		case Calendar.SECOND: return "(" + count + " SEC)";
-		case Calendar.HOUR_OF_DAY: return "(" + count + " HR)";
-		case Calendar.MINUTE: return "(" + count + " MIN)";
-		case Calendar.DAY_OF_MONTH: return "(" + count + " DAY)";
-		case Calendar.MONTH: return "(" + count + " MON)";
-		default: return "(" + count + " calInc=" + calConstant + ")"; 
-		}
-	}
-	/**
+        try
+        {
+            IntervalIncrement [] iia = parseMult(s);
+            return iia[0];
+        }
+        catch(NoSuchObjectException ex)
+        {
+            Logger.instance().warning("IntervalIncrement.parse Invalid Interval: " + ex.getMessage()
+                + " -- using 1 hour");
+            return ONE_HOUR;
+        }
+    }
+
+    /**
+     * Given a string like "h" "hour" "yr" "day" "min", return the
+     * corresponding Calendar constant
+     * @param x the string
+     * @return the Calendar constant or -1 if no match
+     */
+    public static int str2CalConst(String x)
```

Comment:
consider if we could do exact matches.   i.e.    's' or 'seconds' = SECOND

Perhaps define a set of valid string calendar intervals for clarity.

Reply:
except for the toTemporal i added that's all existing code i don't intend to touch at this time.